### PR TITLE
Add entry form and some rules to it

### PR DESCRIPTION
**Pull request by renandvp**

### CODE REVIEW

File: public/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    var user = {name: $("#name").val() , word: $("#word").val() }
+    if(user.name.trim() ==''){
+        alert("Por favor, coloque um nome valido");
+        return;
+    }
+    if(user.word.trim() ==''){
+        alert("Por favor, coloque um palavra valida");
+        return;
+    }
+    socket.emit("start", user, function (cb) {
```
Comment (jeansouzak) [inline public/index.js]: Use emit to notify events.
Don't use to submit form, make a conventional form submit to express server.
I could inject code to server too...

File: public/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    var user = {name: $("#name").val() , word: $("#word").val() }
+    if(user.name.trim() ==''){
+        alert("Por favor, coloque um nome valido");
+        return;
+    }
+    if(user.word.trim() ==''){
+        alert("Por favor, coloque um palavra valida");
+        return;
+    }
+    window.location.href = '/draw?name=' + user.name + '&word=' + user.word;
```
Comment (jeansouzak) [inline public/index.js:14]: Could use POST method to send this

File: public/js/draw.js [hunk 3]
```diff
@@ -20,6 +20,16 @@ document.addEventListener("DOMContentLoaded", function () {
     setupSocket();
     startLoop();
 
+    //Not Changing color, 
+    // var colors = document.getElementsByClassName('color');
```
Comment (jeansouzak) [inline public/js/draw.js:24]: Commented code?

File: public/js/draw.js [hunk 4]
```diff
@@ -41,6 +51,18 @@ document.addEventListener("DOMContentLoaded", function () {
         socket.on('draw', function(line) {
             draw(line);
         });
+
+        $("form#chat").submit(function (e) {
+            e.preventDefault();
+            var guessWord = $(this).find("#guess_word").val();
+            var name = $(this).find("#user_name").val();
+            socket.emit("send word", { guessWord: guessWord, name: name}, function (cb) {
```
Comment (jeansouzak) [inline public/js/draw.js:59]: It is not secure. Use post and handle it on backend.
If you using that other users will see all guess word sended

File: server.js [hunk 5]
```diff
@@ -6,17 +6,49 @@ const io = require('socket.io')(http);
 const port = 5000;
 const host = 'localhost';
 
+var names = [];
```
Comment (jeansouzak) [inline server.js:9]: Why this list?
Name belongs to player object

File: server.js [hunk 6]
```diff
@@ -6,17 +6,49 @@ const io = require('socket.io')(http);
...
+var words = [];
```
Comment (jeansouzak) [inline server.js:10]: think about using let or const instead of var

File: server.js [hunk 7]
```diff
@@ -6,17 +6,49 @@ const io = require('socket.io')(http);
...
+var players = [];
+
+//TODO: User this object to print final point, and give points do the righ user
```
Comment (jeansouzak) [inline server.js:13]: what?

File: server.js [hunk 8]
```diff
@@ -6,17 +6,49 @@ const io = require('socket.io')(http);
...
+var user = {name: null, word: null, points: 0}
+
 app.use(express.static(__dirname + '/public'));
+app.set('views', __dirname + '/public/views');
+app.engine('html', require('ejs').renderFile);
+app.set('view engine', 'html');
 
-//History with drawed lines to render on new client connection
-const drawedList = [];
```
Comment (jeansouzak) [inline server.js:12]: Why you deleted this list?
That is needed to other users can see on start new game or refresh de page
Need to be implemented

File: server.js [hunk 9]
```diff
@@ -6,17 +6,49 @@ const io = require('socket.io')(http);
...
+app.get('/draw', (req, res) => {
+  if (!(names.indexOf(req.query.name) > -1)) {
```
Comment (jeansouzak) [inline server.js:22]: You need validate other needed params

File: server.js [hunk 10]
```diff
@@ -6,17 +6,49 @@ const io = require('socket.io')(http);
...
+    user.name = req.query.name
```
Comment (jeansouzak) [inline server.js:23]: You could use a factory to create an user object

File: server.js [hunk 11]
```diff
@@ -6,17 +6,49 @@ const io = require('socket.io')(http);
...
+    user.word = req.query.word
+    players.push(user);
+    names.push(req.query.name);
```
Comment (jeansouzak) [inline server.js:26]: Why do you use names and words list?
Could you use an unique list liked  with user?

File: server.js [hunk 12]
```diff
@@ -6,17 +6,49 @@ const io = require('socket.io')(http);
...
 
 io.on('error', e => console.log(e));
 
 io.on('connection', function (socket) {
   socket.on('draw', function (line) {
     socket.broadcast.emit('draw', line);
-  });
+  })
+
+  socket.on('send word', function(obj, callback){
```
Comment (jeansouzak) [inline server.js:42]: You could use an express endpoint to receive this word and not traffic with socket
After you receive this word you could compare to players word sended and emit points if is match
You could emit event to display message in global chat after that as well.

File: server.js [hunk 13]
```diff
@@ -6,17 +6,49 @@ const io = require('socket.io')(http);
...
+    if((words.indexOf(obj.guessWord) > -1)){
+      var thisUser = players.find(user => user.name = obj.name);
+      thisUser.points++;
+      socket.emit('send points', thisUser)
+      callback(true);
```
Comment (jeansouzak) [inline server.js:47]: What is this man?????
callback?
Not use that instead use http request to know the response I commented about that above

**Comment by renandvp** [comment]
Too many mistakes, going to redo everything

**Pull request closed without merging**